### ISSUE TITLE
test error

### ISSUE THREAD

**Issue by xuefenfei712** [JoeKawash/ARIADNA#1]
Dear author
Thanks for your ARIADNA, I try with your test data and got the error

 python ARIADNA.py test/test.bam test/test.fasta validate/trail.vcf
Translocations before filter: 0
Translocations after filter: 0
Error: please check bam reference and output

Can I know what's wrong with it?
Best
Xue

**Comment by JoeKawash** [comment]
Hi Xue,

Happy to help. Lets start with which version of python and scikit-learn are you using? ARIADNA needs python 2.7 and scikit-learn 0.18.2. 

-Joe

**Comment by xuefenfei712** [comment]
Thankyou Joe, I use python1.7 and scikit-learn 0.20.4, I cannot founf the old version, it updated to 1.0
Best
Xue

**Comment by JoeKawash** [comment]
Hi Xue,

Older versions are still hosted, you can try something like sudo python2.7 -m pip install scikit-learn==0.18.2 . I have just tested that this still works, but I see that it should be updated and will be doing that soon. 

-Joe

**Comment by xuefenfei712** [comment]
Thankyou for your quick reply, I had the result, though I got a lot of warnings, 
/py2/lib/python2.7/site-packages/sklearn/utils/validation.py:395: DeprecationWarning: Passing 1d arrays as data is deprecated in 0.17 and will raise ValueError in 0.19. Reshape your data either using X.reshape(-1, 1) if your data has a single feature or X.reshape(1, -1) if it contains a single sample.
  DeprecationWarning)

I used my own data, it whoed below:
py2/lib/python2.7/site-packages/sklearn/ensemble/weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.
  from numpy.core.umath_tests import inner1d
Calling variants
rm: impossible de supprimer GRI1TD.tmp.txt.ctx: Aucun fichier ou dossier de ce type
Thankyou
Best
Xue

**Comment by JoeKawash** [comment]
For now the depreciation warnings can be ignored. It seems that the test set did work, therefore we need to know more about your own data. Do you have the required reference (in fasta format), .bam (that was aligned to the same reference), and .bai files? How large is your genome? Are you using PE reads? What is the read depth? Feel free to email me if that is an easier way to share details of your data.

**Comment by xuefenfei712** [comment]
Here is the link
https://drive.google.com/drive/folders/1Fn6w8lEQYGUV4GF1IrhwsNiVmpM3Qfj8?usp=sharing
Thankyou!

**Comment by JoeKawash** [comment]
Hi Xue,

I have tested your data and did not encounter the problem that you list above. However, your .bam seems oddly formatted and only contains a single pair of reads, the rest are missing pairs? The provided .bai file does not match this truncated .bam file. Using software such as samtools, the variants that it finds are not the same ones that you have provided in test1.vcf, we would expect that at least a large number of them would be matching. I had to pull the reads from your .bam file and re-align them to your reference and generate new .bam/.bai files. 100kb is a bit small of a test set, I recommend using 1 or 2 MB of data would be better. 

I suggest creating a new test set using a ~2MB truncated section of chromosome (or a small chromosome), and aligning your original reads there, instead of cutting an existing .bam file. Do not forget to sort and index this new .bam file. This would provide a better base for testing and comparison.

[ARIADNA_XUE_TESTING.zip](https://github.com/JoeKawash/ARIADNA/files/7855656/ARIADNA_XUE_TESTING.zip)

**Comment by xuefenfei712** [comment]
Thankyou so much, I also remap it, and it works, quite useful.
I notice that the final vcf file has lower SNPs than the validated vcf, actually I want to know the genotype of specific loci, even it is the same with the reference genome. Then lower depth <3 need to be marked. 
Can I get this?
Best
Xue

**Comment by JoeKawash** [comment]
The program was designed to filter for only high quality SNPs in ancient data, so it is expected that it will have fewer calls than other software. Because of the way it collects the data for calling, it will only work on larger genome segments, therefore it is best to parse the output for the specific loci you are interested in and check for presence of a SNP. Read depth across the genome can easily be found using other tools.
